src/allocator.hpp

```diff
@@ -21,9 +20,9 @@ namespace loops
         void protect2Execution(uint8_t* a_buffer);
         static Allocator* getInstance(); //TODO(ch): I hope, it's temporary solution. And definetely non-threadsafe.
     private:
+        Allocator();
         size_t m_cachedSize; //TODO(ch): element of disposable implementation.
```

Comment:
there is a serious design flaw here. Allocator in your design is a singleton. Therefore, each class member is a global variable. It means that the current Loops is not thread-safe. Please, revise API. Get rid of global variable.

Reply:
the problem with thread safety is not solved yet, right? please, solve it

Reply:
Done.